snuba/optimize.py

```diff
@@ -169,49 +314,49 @@ def optimize_partition_runner(
                         clickhouse,
                         database,
                         table,
-                        divided_parts[i],
-                        ignore_cutoff,
+                        divided_partitions[i],
+                        cutoff_time,
+                        tracker,
                         clickhouse_host,
                     ),
                 )
             )
 
             threads[i].start()
 
+        # Wait for all threads to finish. Any thread can raise
+        # JobTimeoutException. Its an indication that not all partitions
+        # could be optimized in the given amount of threaded_cutoff_time.
         for i in range(0, parallel):
-            threads[i].join()
+            try:
+                threads[i].join()
+            except JobTimeoutException:
+                pass
```

Comment:
Doesn't this make the JobTimeoutException completely transparent? 
We should have the job crash loudly if the job goes past its overall cutoff time so we can alert on it.